### PR TITLE
Sc render applied

### PR DESCRIPTION
**Pull request by Jinyu-W**
# Description

- Update BE with several minor logic & bug fixes
- Update SC test module
- Revert maro utils logger (v1)
- Update plant & super_vendor topology config files
- Add render and evaluation dump logic to SC example
- SC RL example several bug fixes
- Update to fit RLComponentBundle

## Linked issue(s)/Pull request(s)

- [issue_number](issue_link)

## Type of Change

- [ ] Non-breaking bug fix
- [ ] Breaking bug fix
- [ ] New feature
- [ ] Test
- [ ] Doc update
- [ ] Docker update

## Related Component

- [ ] Simulation toolkit
- [ ] RL toolkit
- [ ] Distributed toolkit

## Has Been Tested

- OS:
  - [ ] Windows
  - [ ] Mac OS
  - [ ] Linux
- Python version:
  - [ ] 3.6
  - [ ] 3.7
- Key information snapshot(s):

## Needs Follow Up Actions

- [ ] New release package
- [ ] New docker image

## Checklist

- [ ] Add/update the related comments
- [ ] Add/update the related tests
- [ ] Add/update the related documentations
- [ ] Update the dependent downstream modules usage

### CODE REVIEW

File: examples/supply_chain/common/balance_calculator.py [hunk 1]
```diff
@@ -1,50 +1,39 @@
...
+class GlobalProductInfo:
+    unit_id: int
+    sku_id: int
+    node_index: int
+    storage_index: Optional[int]
+    distribution_index: Optional[int]
+    consumer_index: Optional[int]
+    seller_index: Optional[int]
+    manufacture_index: Optional[int]
+    downstream_product_id_list: List[int]
```
Comment (lihuoran) [inline examples/supply_chain/common/balance_calculator.py:29]: Shall this one be changed to `sku_id`?

There are still many `product_id` in the project.

File: examples/supply_chain/common/utils.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+def _check_attribute_keys(env: Env, target_type: str, attribute: str) -> None:
+    valid_target_types = set(env.summary["node_detail"].keys())
+    assert target_type in valid_target_types, f"Target_type {target_type} not in {list(valid_target_types)}!"
+
+    valid_attributes = set(env.summary["node_detail"][target_type]["attributes"].keys())
+    assert attribute in valid_attributes, (
+        f"Attribute {attribute} not valid for {target_type}. Valid attributes: {list(valid_attributes)}"
+    )
+
+def get_attributes(env: Env, target_type: str, attribute: str, tick: Optional[int]) -> np.ndarray:
```
[the remark this answers is not in the document]
Comment (Jinyu-W) [reply examples/supply_chain/common/utils.py:20]: yes

File: examples/supply_chain/common/utils.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    if tick is None:
+        num_frame = env.frame_index + 1
+        num_nodes = len(env.snapshot_list[target_type])
+        return env.snapshot_list[target_type][::attribute].reshape(num_frame, num_nodes)
+
+    else:
+        frame_index = env.business_engine.frame_index(tick)
+        return env.snapshot_list[target_type][frame_index::attribute].flatten()
+
+def get_list_attributes(
```
[the remark this answers is not in the document]
Comment (Jinyu-W) [reply examples/supply_chain/common/utils.py:32]: List[1-D matrix] & List[List[1-D matrix]]

File: examples/supply_chain/common/render_tools/comparision_html_render.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+manufacture_finished_quantity,
+manufacture_manufacture_cost,product_price,
+product_check_in_quantity_in_order,product_delay_order_penalty,
+product_transportation_cost
+distribution_pending_product_quantity,
+distribution_pending_order_number
+"""
+
+
+class SimulationComparisionTrackerHtml:
```
[the remark this answers is not in the document]
Comment (Jinyu-W) [reply examples/supply_chain/common/render_tools/comparision_html_render.py:34]: Not used in the workflow now, leave it for future refactoring.

File: maro/rl/rollout/env_sampler.py [hunk 5]
```diff
@@ -382,18 +382,19 @@ def sample(self, policy_state: Optional[Dict[str, dict]] = None, num_steps: Opti
                     for id_, env_action in env_action_dict.items() if id_ in self._trainable_agents
                 },
             )
+
+            # Update env and get new states (global & agent)
```
[the remark this answers is not in the document]
Comment (Jinyu-W) [reply maro/rl/rollout/env_sampler.py:386]: to calculate the reward correspondinging to the newest state & action.

File: maro/rl/training/algorithms/base/ac_ppo_base.py [hunk 6]
```diff
@@ -144,7 +144,7 @@ def _get_actor_loss(self, batch: TransitionBatch) -> Tuple[torch.Tensor, bool]:
             kl = (logps_old - logps).mean().item()
             early_stop = (kl >= 0.01 * 1.5)  # TODO
             clipped_ratio = torch.clamp(ratio, 1 - self._clip_ratio, 1 + self._clip_ratio)
-            actor_loss = -(torch.min(ratio * advantages, clipped_ratio * advantages)).mean()
+            actor_loss = -(torch.min(ratio * advantages, clipped_ratio * advantages))
```
[the remark this answers is not in the document]
Comment (Jinyu-W) [reply maro/rl/training/algorithms/base/ac_ppo_base.py:147]: typo, fixed

File: maro/rl/training/algorithms/dqn.py [hunk 7]
```diff
@@ -67,6 +67,7 @@ def __init__(
         self._target_policy: ValueBasedPolicy = clone(self._policy)
         self._target_policy.set_name(f"target_{self._policy.name}")
         self._target_policy.eval()
+        self._device = None
```
[the remark this answers is not in the document]
Comment (Jinyu-W) [reply maro/rl/training/algorithms/dqn.py:70]: fixed

File: maro/rl/training/algorithms/ppo.py [hunk 8]
```diff
@@ -33,6 +40,127 @@ def __post_init__(self) -> None:
         assert self.clip_ratio is not None
 
 
+class DiscretePPOOps(ACBasedOps):
```
[the remark this answers is not in the document]
Comment (Jinyu-W) [reply maro/rl/training/algorithms/ppo.py:43]: renamed

File: examples/supply_chain/common/balance_calculator.py (hunk 1 above)
Comment (Jinyu-W) [reply examples/supply_chain/common/balance_calculator.py:29]: product_id means product unit id here.
